### PR TITLE
Solves bug when calling `ConjugateGradientPoissonSolver` without specifying preconditioner

### PR DESCRIPTION
**Pull request by tomchor**
See https://github.com/CliMA/Oceananigans.jl/issues/3831#issuecomment-2406235013

Solves https://github.com/CliMA/Oceananigans.jl/issues/3829

### CODE REVIEW

File: src/Solvers/conjugate_gradient_poisson_solver.jl [hunk 1]
```diff
@@ -1,4 +1,5 @@
 using Oceananigans.Operators: divᶜᶜᶜ, ∇²ᶜᶜᶜ 
+using Oceananigans.Grids: ImmersedBoundaryGrid
```
Comment (glwagner) [inline src/Solvers/conjugate_gradient_poisson_solver.jl:2]: Hmm, I think `ImmersedBoundaryGrid` is defined by `ImmersedBoundaries`, which isn't imported until after `Solvers`:

https://github.com/CliMA/Oceananigans.jl/blob/99d3c1fcb8cb7951c0b82f82d9d61d2fe9d541e1/src/Oceananigans.jl#L210-L227

A bit of refactoring is needed to import `ImmersedBoundaries` before `Solvers`. So I think if we want to resolve this without all that refactoring we should define the constructor that depends on `ImmersedBoundaryGrid` separately (either in `ImmersedBoundaries` or `Model.NonhydrostaticModels`)

**Comment by glwagner** [comment]
I can fix this if you want

File: src/Solvers/conjugate_gradient_poisson_solver.jl (hunk 1 above)
Comment (tomchor) [reply src/Solvers/conjugate_gradient_poisson_solver.jl:2]: yeah I think you're right!

**Comment by tomchor** [comment]
> I can fix this if you want

Thanks, yeah feel free to jump in!

**Comment by ali-ramadhan** [comment]
This might be a duplicate of PR #3830 but looks like both will be superseded by PR #3847 which is doing the refactoring necessary for this fix.

**Review by ali-ramadhan — APPROVED** [review]
Thanks @tomchor!

PS: You can mark this PR as resolving https://github.com/CliMA/Oceananigans.jl/issues/3829

**Comment by tomchor** [comment]
> Thanks @tomchor!

I mean, the real hero was @simone-silvestri with https://github.com/CliMA/Oceananigans.jl/pull/3847 :)

> PS: You can mark this PR as resolving #3829

Just did!

**Pull request merged**